pkg/common/middleware/authorization.go

```diff
@@ -0,0 +1,80 @@
+package middleware
+
+import (
+	"fmt"
+	"net/http"
+	"strings"
+
+	"github.com/gofiber/fiber/v2"
+	log "github.com/sirupsen/logrus"
+
+	"github.com/G-Research/fasttrackml/pkg/common/api"
+)
+
+// NewRBACAuthorizationMiddleware creates new RBAC based Middleware instance.
+func NewRBACAuthorizationMiddleware(config map[string]map[string]struct{}) fiber.Handler {
```

Comment:
this could probably just be `NewRBACMiddleware` since the 'AC' kind of implies auth